### PR TITLE
[IMP] fieldservice_recurring: populate team_id from order template

### PR DESCRIPTION
**Pull request by ivantodorovich**

### CODE REVIEW

File: fieldservice_sale_recurring/models/sale_order_line.py [hunk 1]
```diff
@@ -20,21 +20,24 @@ def _field_create_fsm_recurring_prepare_values(self):
...
+        vals = {
             "location_id": self.order_id.fsm_location_id.id,
             "start_date": self.order_id.expected_date,
             "fsm_recurring_template_id": template.id,
             "description": note,
             "max_orders": template.max_orders,
             "fsm_frequency_set_id": template.fsm_frequency_set_id.id,
-            "fsm_order_template_id": product.fsm_order_template_id.id
-            or template.fsm_order_template_id.id,
+            "fsm_order_template_id": order_template.id,
```
Comment (max3903) [inline fieldservice_sale_recurring/models/sale_order_line.py:34]:
```suggestion
            "fsm_order_template_id": order_template.id,
            "team_id": order_template.team_id.id or False,
```

File: fieldservice_sale_recurring/models/sale_order_line.py [hunk 2]
```diff
@@ -20,21 +20,24 @@ def _field_create_fsm_recurring_prepare_values(self):
...
             "sale_line_id": self.id,
             "company_id": self.company_id.id,
         }
+        if order_template.team_id:
+            vals["team_id"] = order_template.team_id.id
```
Comment (max3903) [inline fieldservice_sale_recurring/models/sale_order_line.py:39]:
```suggestion
```

File: fieldservice_recurring/models/fsm_recurring.py [hunk 3]
```diff
@@ -107,14 +107,17 @@ def onchange_recurring_template_id(self):
     def populate_from_template(self, template=False):
         if not template:
             template = self.fsm_recurring_template_id
-        return {
+        vals = {
             "fsm_frequency_set_id": template.fsm_frequency_set_id,
             "max_orders": template.max_orders,
             "description": template.description,
             "fsm_order_template_id": template.fsm_order_template_id,
```
Comment (max3903) [inline fieldservice_recurring/models/fsm_recurring.py:114]:
```suggestion
            "fsm_order_template_id": template.fsm_order_template_id,
            "team_id": template.fsm_order_template_id.team_id and template.fsm_order_template_id.team_id.id or False
```

File: fieldservice_recurring/models/fsm_recurring.py [hunk 4]
```diff
@@ -107,14 +107,17 @@ def onchange_recurring_template_id(self):
...
             "scheduled_duration": template.fsm_order_template_id.duration,
             "company_id": template.company_id,
         }
+        if template.fsm_order_template_id.team_id:
+            vals["team_id"] = template.fsm_order_template_id.team_id.id
```
Comment (max3903) [inline fieldservice_recurring/models/fsm_recurring.py:119]:
```suggestion
```

File: fieldservice_recurring/models/fsm_recurring.py (hunk 3 above)
Comment (ivantodorovich) [reply fieldservice_recurring/models/fsm_recurring.py:114]: The `team_id` field on the template is optional, whilst the `team_id` on the order is mandatory.
This is why I chose to not return an explicit `False` value here, and so let the field default take over if empty.

File: fieldservice_recurring/models/fsm_recurring.py (hunk 4 above)
Comment (ivantodorovich) [reply fieldservice_recurring/models/fsm_recurring.py:119]: See previous comment

File: fieldservice_sale_recurring/models/sale_order_line.py (hunk 1 above)
Comment (ivantodorovich) [reply fieldservice_sale_recurring/models/sale_order_line.py:34]: See previous comment. The same applies for the `fsm.recurring` model, whose `team_id` is required.

File: fieldservice_sale_recurring/models/sale_order_line.py (hunk 2 above)
Comment (ivantodorovich) [reply fieldservice_sale_recurring/models/sale_order_line.py:39]: same

**Review by yankinmax — APPROVED** [review]

**Review by max3903 — APPROVED** [review]

**Review by brian10048 — APPROVED** [review]

**Pull request merged**